Source/Core/VideoBackends/Vulkan/VKTexture.cpp

```diff
@@ -752,6 +774,75 @@ std::unique_ptr<VKStagingTexture> VKStagingTexture::Create(StagingTextureType ty
   return staging_tex;
 }
 
+std::pair<VkImage, VkDeviceMemory> VKStagingTexture::CreateLinearImage(StagingTextureType type,
+                                                                       const TextureConfig& config)
+{
+  // Create a intermediate texture with linear tiling
+  VkImageCreateInfo image_info = {VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO,
+                                  nullptr,
+                                  0,
+                                  VK_IMAGE_TYPE_2D,
+                                  VKTexture::GetVkFormatForHostTextureFormat(config.format),
+                                  {config.width, config.height, 1},
+                                  1,
+                                  1,
```

Comment:
Should this be `config.levels`? I don't 100% know how the staging texture system works, so if this is supposed to be only one layer, that's fine. But I'd recommend testing with stereoscopic 3D enabled to make sure this behaves correctly.

Reply:
If you're testing stereoscopic 3d, you probably should do this on a non phone device.  Supposedly stereoscopic 3d doesn't work on some (all?) devices, see issue 12147.  Regardless @Pokechu22 's suggestion seems reasonable to me, though it'd be `config.layers` not levels I believe.

Reply:
Oops, yeah, I meant `config.layers`. I think this code is only used with the new `BUG_SLOW_OPTIMAL_IMAGE_TO_BUFFER_COPY` which only applies on phones though, so it might not be meaningful to test it at all (or you'd need to force-enable it when testing).